contracts/zerodev/account/kernel/WalletKernel.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "../../plugin/IPlugin.sol";
+import "../../../core/Helpers.sol";
+import "../../../interfaces/IAccount.sol";
+import "../../utils/OpinionatedExec.sol";
+import "./Compatibility.sol";
+
+struct WalletKernelStorage {
+    address owner;
+    uint256 nonce;
+}
+
+
+/// @title Wallet Kernel
+/// @author taek<leekt216@gmail.com>
+/// @notice wallet kernel for minimal wallet functionality
+/// @dev supports only 1 owner and 1 threshold, multiple plugins
+contract WalletKernel is IAccount, EIP712, Compatibility {
+
+    function getWalletKernelStorage() internal pure returns (WalletKernelStorage storage ws) {
+        bytes32 storagePosition = bytes32(uint256(keccak256("zero-dev.account.walletkernel")) - 1);
```

Comment:
Do we expect people to be inheriting from this kernel?  If so, would it make sense to enable people to set their own prefix?

Reply:
Hmmm 🤔 I get what you mean so we can let them use their own storage slot right?
I think this part is quite dependant on the entrypoint upgrade but i think it's good to have one slot that is shared by all inherited contracts. This way, all wallet that inherits Kernel does not have to deal with storage slot migration on this core parts